src/server/command_registry.h

```diff
@@ -123,6 +143,24 @@ class CommandRegistry {
       cb(k_v.first, k_v.second);
     }
   }
+
+  void ResetCallStats(unsigned thread_index) {
+    for (auto& k_v : cmd_map_) {
+      k_v.second.ResetStats(thread_index);
+    }
+  }
+
+  template <typename Map> void MergeCallStats(unsigned thread_index, Map* dest) const {
+    for (const auto& k_v : cmd_map_) {
+      auto src = k_v.second.GetStats(thread_index);
+      if (src.first == 0)
+        continue;
+
+      auto& ent = (*dest)[typename Map::key_type{k_v.first}];
+      ent.first += src.first;
+      ent.second += src.second;
+    }
+  }
```

Comment:
I thought you were not a fan of big header based implementations, also not sure why the template if you use it only once - to avoid naming the type? then maybe you can just accept a function to avoid naming it, its called rarely anyways